src/numeric/decision_rules.jl

```diff
@@ -1,9 +1,43 @@
-import splines
+abstract AbstractDecisionRule{S,T}
 
-abstract AbstractDecisionRule
-abstract AbtsractCubicSplineDR <: AbstractDecisionRule
+# abstract AbstractCachedDecisionRule{S,T} <: AbstractDecisionRule{S,T}
+# we don't implement that using subtypes anymore
+
+type DecisionRule{S,T} <: AbstractDecisionRule{S,T}
+    grid_exo::S
+    grid_endo::T
+    n_x::Int # number of values (could be a method if there was a nice method name)
+    coefficients::Array{Array{Float64}}
```

Comment:
We don't know generically but we know for a given model. The element `coefficients[1]` has dimension `n_x \time n_1 ... \times n_d`. This dimension is used to dispatch to the right filtering/interpolating functions. I guess you could parameterize everything by the number of dimensions (maybe parameterize Grid), but only if it makes a sizable performance gain.